waiter/src/waiter/core.clj

```diff
@@ -219,12 +219,20 @@
         handler)))
 
 (defn wrap-debug
-  "Attaches debugging headers to requests when enabled."
+  "Attaches debugging headers to requests when enabled.
+   Logs any request trailers when they are provided."
   [handler generate-log-url-fn]
   (fn wrap-debug-fn
-    [{:keys [client-protocol internal-protocol request-id request-time router-id] :as request}]
+    [{:keys [client-protocol internal-protocol request-id request-time router-id trailers-fn] :as request}]
     (if (utils/request->debug-enabled? request)
-      (let [response (handler request)
+      (let [request (cond-> request
```

Comment:
This `request` is shadowing the one in the argument list. Do we have a convention for shadowing names?

Reply:
My rule of thumb has been to shadow when I know I no longer want access to the old one, it also helps avoid programmer error.

We have been using ticks (`'`) in the loops when we want access to the old and new values.